erroranalysis/erroranalysis/_internal/surrogate_error_tree.py

```diff
@@ -739,6 +741,34 @@ def node_to_dict(df, tree, nodeid, categories, json,
     return json, df
 
 
+def filter_by_threshold(df, p_node_name_val, parent_threshold, side):
+    """Filters the DataFrame by the threshold of the parent node.
+
+    :param df: The DataFrame to filter.
+    :type df: pandas.DataFrame
+    :param p_node_name_val: The name of the parent node.
+    :type p_node_name_val: str
+    :param parent_threshold: The threshold of the parent node.
+    :type parent_threshold: float
+    :param side: The side of the parent node.
+    :type side: TreeSide
+    """
+    try:
+        if side == TreeSide.LEFT_CHILD:
+            df = df[df[p_node_name_val] <= parent_threshold]
+        else:
+            df = df[df[p_node_name_val] > parent_threshold]
+    except TypeError as e:
+        mismatch_err = "not supported between instances of 'str' and 'float'"
```

Comment:
Is there a better way to check for this error? It might be really flaky to check for the above error message. If the underlying message changes then this logic will break.

Reply:
May be check for the dtype of the that column. it shouldn't be integer or float in this case.

Reply:
Done.  The dtype is object in this case (and integer values can have object dtype as well), so I checked the value manually instead.